secure_message/repository/retriever.py

```diff
@@ -155,11 +156,16 @@ def _retrieve_internal_thread_list(request_args, user):
                                  .label('max_id')) \
                 .join(Events).join(Status) \
                 .filter(or_(and_(SecureMessage.from_internal.is_(False), Status.label == Labels.INBOX.value),  # NOQA
-                            and_(SecureMessage.from_internal.is_(True),
-                                 Status.label.in_([Labels.SENT.value]))
-                           )
-                       ) \
-                .group_by(SecureMessage.thread_id).subquery('t')
+                            and_(SecureMessage.from_internal.is_(True), Status.label == Labels.SENT.value))
+                        ).group_by(SecureMessage.thread_id).subquery('t')
+
+            u = db.session.query(SecureMessage.thread_id).join(Conversation) \
+                .filter(Conversation.is_closed.is_(True)).subquery('u')
+
+            if request_args.label == 'CLOSED':
+                conditions.append(SecureMessage.thread_id.in_(u))
+            else:
+                conditions.append(~SecureMessage.thread_id.in_(u))
```

Comment:
I think this change should be done as part of the change for the listing. If we do this without updating the frontend, won't the closed messages disappear from the listing?

Reply:
Thats fair.  I'll make a closed-conversation-only branch that won't have these changes in it